cli/ops/tty.rs

```diff
@@ -250,3 +251,80 @@ pub fn op_isatty(
     })?;
   Ok(JsonOp::Sync(json!(isatty)))
 }
+
+#[derive(Deserialize)]
+struct ConsoleSizeArgs {
+  rid: u32,
+}
+
+#[derive(Serialize)]
+struct ConsoleSize {
+  columns: u32,
+  rows: u32,
+}
+
+pub fn op_console_size(
+  isolate_state: &mut CoreIsolateState,
+  state: &State,
+  args: Value,
+  _zero_copy: &mut [ZeroCopyBuf],
+) -> Result<JsonOp, OpError> {
+  state.check_unstable("Deno.consoleSize");
+  let args: ConsoleSizeArgs = serde_json::from_value(args)?;
+  let rid = args.rid;
+
+  let mut resource_table = isolate_state.resource_table.borrow_mut();
+  let size =
+    std_file_resource(&mut resource_table, rid as u32, move |r| match r {
+      Ok(std_file) => {
+        #[cfg(windows)]
+        {
+          use std::os::windows::io::AsRawHandle;
+          let handle = std_file.as_raw_handle();
+
+          unsafe {
+            let mut bufinfo: winapi::um::wincon::CONSOLE_SCREEN_BUFFER_INFO =
+              std::mem::zeroed();
+
+            if winapi::um::wincon::GetConsoleScreenBufferInfo(
+              handle,
+              &mut bufinfo,
+            ) == 0
+            {
+              // TODO (caspervonb) use GetLastError
+              return Err(OpError::other("windows error".to_owned()));
+            }
+
+            Ok(ConsoleSize {
+              columns: bufinfo.dwSize.X as u32,
+              rows: bufinfo.dwSize.Y as u32,
+            })
+          }
+        }
+
+        #[cfg(unix)]
+        {
+          use std::os::unix::io::AsRawFd;
+
+          let fd = std_file.as_raw_fd();
+          unsafe {
+            let mut size: libc::winsize = std::mem::zeroed();
+            if libc::ioctl(fd, libc::TIOCGWINSZ, &mut size as *mut _) != 0 {
+              // TODO(caspervonb) this should just be OpError::from(errno)
+              // but libc does not provide errno.
+              return Err(OpError::other("ioctl error".to_owned()));
```

Comment:
Can you use last_os_error? https://doc.rust-lang.org/std/io/struct.Error.html#method.last_os_error